draft-irtf-cfrg-vdaf.md

```diff
@@ -1363,6 +1376,57 @@ class PrgSha3(Prg):
 ~~~
 {: title="Definition of PRG PrgSha3."}
 
+### PrgFixedKeyAes128 {#prg-fixed-key-aes128}
+
+While PrgSha3 as described above can be securely used in all cases where a Prg
+is needed in the VDAFs described in this document, there are some cases where
+a more efficient instantiation based on fixed-key AES is possible. For now, this
+is limited to the Prg used inside the Idpf {{idpf}} implementation in Poplar1
+{{idpf-poplar}}. It is NOT RECOMMENDED to use this Prg anywhere else.
+See Security Considerations {{security}} for a more detailed discussion.
+
+~~~
+class PrgFixedKeyAes128(Prg):
+    # Associated parameters
+    SEED_SIZE = 16
+
+    def __init__(self, seed, custom, binder):
+        self.length_consumed = 0
+
+        # Use SHA-3 to derive a key from the binder and customization
+        # strings. Note that the AES key does not need to be kept
+        # secret from any party. However, when used with IpdfPoplar,
+        # we require the binder to be a random nonce.
+        #
+        # Implementation note: This step can be cached across PRG
+        # evaluations with many different seeds.
+        self.fixed_key = cSHAKE128(binder, 16, b'', custom)
+        self.seed = seed
+
+    def next(self, length: Unsigned) -> Bytes:
+        offset = self.length_consumed % 16
+        new_length = self.length_consumed + length
+        block_range = range(
+            int(self.length_consumed / 16),
+            int(new_length / 16) + 1)
+        self.length_consumed = new_length
+
+        hashed_blocks = [
+            self.hash_block(xor(self.seed, to_le_bytes(i, 16))) \
+                         for i in block_range
+        ]
+        return concat(hashed_blocks)[offset:offset+length]
+
+    # The multi-instance tweakable circular correlation-robust hash function of
+    # [GKWWY20] (Section 4.2).
```

Comment:
I don't see any tweaking going on here.

Reply:
cc/ @schoppmp

Reply:
If you look at Section 4.2 in https://eprint.iacr.org/2019/1168.pdf, the tweak i is passed as the key to the block cipher. So for a single client, the tweak stays the same, but it differs across clients.